app/assets/javascripts/cookiebanner.js

```diff
@@ -0,0 +1,110 @@
+        /*  This callback is called when the 'accept' action is fired within the cookie banner
+        *   This is where you'd hide the first stage in a decision/confirmation style banner
+        */
+        const bannerAcceptCallback = function () {
+            document.querySelector('.cm-cookie-banner__decision').hidden = true;
+            document.querySelector('.cm-cookie-banner__confirmation').hidden = false;
+        }
+
+        /*  This callback is called when the 'reject' action is fired within the cookie banner
+        *   This is where you'd hide the first stage in a decision/confirmation style banner
+        */
+        const bannerRejectCallback = function () {
+            document.querySelector('.cm-cookie-banner__decision').hidden = true;
+            document.querySelector('.cm-cookie-banner__confirmation').hidden = false;
+        }
+
+        /*  This callback is called when the banner save action is fired within the cookie banner (caused by accepting or rejecting).
+        *   This is where you'd likely call a method to that would programmatically remove or add the
+        *   consent-required scripts/tags back onto the page.
+        */
+        const bannerSaveCallback = function (cookieStatus) {
+            updateConsent(cookieStatus);
+        }
+
+        /*  This callback is called when the save action is fired within the preference form (caused by saving the form).
+        *   This is where you'd likely call a method to that would programmatically remove or add the
+        *   consent-required scripts/tags back onto the page.
+        */
+        const preferenceFormSaveCallBack = function (cookieStatus) {
+            updateConsent(cookieStatus);
+        }
+
+        /*  This is where you'd perform logic regarding granting or denying the injection
+        *   of cookies, scripts and tags that required consent. This callback is called
+        *   with an object containing the current cookie preferences.
+        */
+        const updateConsent = function (cookieStatus) {
+            // Google Tag Manager Analytics Consent
+            const gtag = window.gtag || function () {
+                dataLayer.push(arguments);
+            };
+            if (cookieStatus.analytics === 'on') {
+                gtag('consent', 'update', {'analytics_storage': 'granted'});
+            } else {
+                gtag('consent', 'update', {'analytics_storage': 'denied'});
+            }            // Dynatrace RUM Consent
```

Comment:
As Probate is using GTM for its Google Analytics implementation, you'll need to switch to using a data-layer event that acts as an exception trigger with GTM on GA events. The code below should provide GTM with the latest version of the cookie preferences selected by the user, which GTM can then interpret and use to block further events.
 
```
  // GTM based GA consent
  const dataLayer = window.dataLayer || [];
  dataLayer.push({'event': 'Cookie Preferences', 'cookiePreferences': cookieStatus});
```